### PR TITLE
Detect and use gindent if available

### LINKED ISSUE #171: Use gindent if available in indent.sh
This will make it so that `indent.sh` does not have to be modified to be run on MacOS.

### PR DESCRIPTION
**Pull request by ryanplusplus**
Closes #171

### CODE REVIEW

**Comment by patricksjackson** [comment]
Should this be accompanied by any change to the README?

I don't have a machine to test this with, but it looks fine.

**Comment by ryanplusplus** [comment]
@patricksjackson good call. Notes added to `README.md`.

File: README.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ Each exercise should be contained in a directory ```c/exercises/<my exercise>```
...
+* `src` - contains the example files ```example.c``` and ```example.h```. These are both skipped by the ```exercism``` cli when downloading to the client, so it is imperative that you do not reference the names of the files in your code. If you need to provide a header file example that is necessary to run your tests it should be named ```<my exercise>.h``` instead.  Please also use [```include```](http://faculty.cs.niu.edu/~mcmahon/CS241/c241man/node90.html) guards in your header files. The tests can be run using the [```bin/run-tests```](https://github.com/exercism/c/blob/master/bin/run-tests) script which will rename the ```example.{c|h}``` files accordingly.
```
Comment (ryanplusplus) [inline README.md]: The GitHub diff view doesn't show it clearly, but all that happened here was that trailing whitespace was automatically removed by my editor.

**Comment by Gamecock** [comment]
This works on OSX, I added a PR with an extra print that might be helpful.

**Comment by Gamecock** [comment]
Stupid question, but I cannot find where this README appears on `exercism.io`, which is why it didn't help me.  Might be because the track is not launched yet.

**Comment by ryanplusplus** [comment]
@Gamecock I'm honestly not sure. The documentation structure has changed a few times recently in the run-up to Exercism v2 and I don't know where stuff goes anymore.

**Comment by patricksjackson** [comment]
That README doesn't show up on exercism.io. It's specifically for just this repo. My understanding of the split is that details about contributing to this track specifically go in that README, details about contributing to the exercism tracks generally can go in the docs repo https://github.com/exercism/docs, and docs for _using_ this track go in https://github.com/exercism/c/tree/master/docs, which show up on exercism.io under this tracks info.

**Pull request merged**